File: fontes/backend/php/fenix/src/model/cte/CteRodoviarioVeiculo.php
<?php
/*******************************************************************************
Title: T2Ti ERP 3.0
Description: Model relacionado à tabela [CTE_RODOVIARIO_VEICULO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
declare(strict_types=1);

class CteRodoviarioVeiculo extends EloquentModel implements \JsonSerializable
{
    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'CTE_RODOVIARIO_VEICULO';

    /**
     * Eager Loading - Relacionamentos que devem ser sempre carregados por padrão
     *
     * @var array
     */
	
    /**
     * Relations
     */

    /**
     * Gets e Sets
     */
    public function getIdAttribute() 
	{
		return $this->attributes['ID'];
	}

	public function setIdAttribute($id) 
	{
		$this->attributes['ID'] = $id;
	}

    public function getIdCteRodoviarioAttribute() 
	{
		return $this->attributes['ID_CTE_RODOVIARIO'];
	}

	public function setIdCteRodoviarioAttribute($idCteRodoviario) 
	{
		$this->attributes['ID_CTE_RODOVIARIO'] = $idCteRodoviario;
	}

    public function getCodigoInternoAttribute() 
	{
		return $this->attributes['CODIGO_INTERNO'];
	}

	public function setCodigoInternoAttribute($codigoInterno) 
	{
		$this->attributes['CODIGO_INTERNO'] = $codigoInterno;
	}

    public function getRenavamAttribute() 
	{
		return $this->attributes['RENAVAM'];
	}

	public function setRenavamAttribute($renavam) 
	{
		$this->attributes['RENAVAM'] = $renavam;
	}

    public function getPlacaAttribute() 
	{
		return $this->attributes['PLACA'];
	}

	public function setPlacaAttribute($placa) 
	{
		$this->attributes['PLACA'] = $placa;
	}

    public function getTaraAttribute() 
	{
		return $this->attributes['TARA'];
	}

	public function setTaraAttribute($tara) 
	{
		$this->attributes['TARA'] = $tara;
	}

    public function getCapacidadeKgAttribute() 
	{
		return $this->attributes['CAPACIDADE_KG'];
	}

	public function setCapacidadeKgAttribute($capacidadeKg) 
	{
		$this->attributes['CAPACIDADE_KG'] = $capacidadeKg;
	}

    public function getCapacidadeM3Attribute() 
	{
		return $this->attributes['CAPACIDADE_M3'];
	}

	public function setCapacidadeM3Attribute($capacidadeM3) 
	{
		$this->attributes['CAPACIDADE_M3'] = $capacidadeM3;
	}

    public function getTipoPropriedadeAttribute() 
	{
		return $this->attributes['TIPO_PROPRIEDADE'];
	}

	public function setTipoPropriedadeAttribute($tipoPropriedade) 
	{
		$this->attributes['TIPO_PROPRIEDADE'] = $tipoPropriedade;
	}

    public function getTipoVeiculoAttribute() 
	{
		return $this->attributes['TIPO_VEICULO'];
	}

	public function setTipoVeiculoAttribute($tipoVeiculo) 
	{
		$this->attributes['TIPO_VEICULO'] = $tipoVeiculo;
	}

    public function getTipoRodadoAttribute() 
	{
		return $this->attributes['TIPO_RODADO'];
	}

	public function setTipoRodadoAttribute($tipoRodado) 
	{
		$this->attributes['TIPO_RODADO'] = $tipoRodado;
	}

    public function getTipoCarroceriaAttribute() 
	{
		return $this->attributes['TIPO_CARROCERIA'];
	}

	public function setTipoCarroceriaAttribute($tipoCarroceria) 
	{
		$this->attributes['TIPO_CARROCERIA'] = $tipoCarroceria;
	}

    public function getUfAttribute() 
	{
		return $this->attributes['UF'];
	}

	public function setUfAttribute($uf) 
	{
		$this->attributes['UF'] = $uf;
	}

    public function getProprietarioCpfAttribute() 
	{
		return $this->attributes['PROPRIETARIO_CPF'];
	}

	public function setProprietarioCpfAttribute($proprietarioCpf) 
	{
		$this->attributes['PROPRIETARIO_CPF'] = $proprietarioCpf;
	}

    public function getProprietarioCnpjAttribute() 
	{
		return $this->attributes['PROPRIETARIO_CNPJ'];
	}

	public function setProprietarioCnpjAttribute($proprietarioCnpj) 
	{
		$this->attributes['PROPRIETARIO_CNPJ'] = $proprietarioCnpj;
	}

    public function getProprietarioRntrcAttribute() 
	{
		return $this->attributes['PROPRIETARIO_RNTRC'];
	}

	public function setProprietarioRntrcAttribute($proprietarioRntrc) 
	{
		$this->attributes['PROPRIETARIO_RNTRC'] = $proprietarioRntrc;
	}

    public function getProprietarioNomeAttribute() 
	{
		return $this->attributes['PROPRIETARIO_NOME'];
	}

	public function setProprietarioNomeAttribute($proprietarioNome) 
	{
		$this->attributes['PROPRIETARIO_NOME'] = $proprietarioNome;
	}

    public function getProprietarioIeAttribute() 
	{
		return $this->attributes['PROPRIETARIO_IE'];
	}

	public function setProprietarioIeAttribute($proprietarioIe) 
	{
		$this->attributes['PROPRIETARIO_IE'] = $proprietarioIe;
	}

    public function getProprietarioUfAttribute() 
	{
		return $this->attributes['PROPRIETARIO_UF'];
	}

	public function setProprietarioUfAttribute($proprietarioUf) 
	{
		$this->attributes['PROPRIETARIO_UF'] = $proprietarioUf;
	}

    public function getProprietarioTipoAttribute() 
	{
		return $this->attributes['PROPRIETARIO_TIPO'];
	}

	public function setProprietarioTipoAttribute($proprietarioTipo) 
	{
		$this->attributes['PROPRIETARIO_TIPO'] = $proprietarioTipo;
	}


    /**
     * Mapping
     */
    public function mapear($objeto)
    {
        if (isset($objeto)) {
            isset($objeto->id) ? $this->setIdAttribute($objeto->id) : $this->setIdAttribute(null);

			$this->setIdCteRodoviarioAttribute($objeto->idCteRodoviario);
			$this->setCodigoInternoAttribute($objeto->codigoInterno);
			$this->setRenavamAttribute($objeto->renavam);
			$this->setPlacaAttribute($objeto->placa);
			$this->setTaraAttribute($objeto->tara);
			$this->setCapacidadeKgAttribute($objeto->capacidadeKg);
			$this->setCapacidadeM3Attribute($objeto->capacidadeM3);
			$this->setTipoPropriedadeAttribute($objeto->tipoPropriedade);
			$this->setTipoVeiculoAttribute($objeto->tipoVeiculo);
			$this->setTipoRodadoAttribute($objeto->tipoRodado);
			$this->setTipoCarroceriaAttribute($objeto->tipoCarroceria);
			$this->setUfAttribute($objeto->uf);
			$this->setProprietarioCpfAttribute($objeto->proprietarioCpf);
			$this->setProprietarioCnpjAttribute($objeto->proprietarioCnpj);
			$this->setProprietarioRntrcAttribute($objeto->proprietarioRntrc);
			$this->setProprietarioNomeAttribute($objeto->proprietarioNome);
			$this->setProprietarioIeAttribute($objeto->proprietarioIe);
			$this->setProprietarioUfAttribute($objeto->proprietarioUf);
			$this->setProprietarioTipoAttribute($objeto->proprietarioTipo);

			// vincular objetos
        }
    }

    /**
     * Serialization
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
			'id' => $this->getIdAttribute(),
			'idCteRodoviario' => $this->getIdCteRodoviarioAttribute(),
			'codigoInterno' => $this->getCodigoInternoAttribute(),
			'renavam' => $this->getRenavamAttribute(),
			'placa' => $this->getPlacaAttribute(),
			'tara' => $this->getTaraAttribute(),
			'capacidadeKg' => $this->getCapacidadeKgAttribute(),
			'capacidadeM3' => $this->getCapacidadeM3Attribute(),
			'tipoPropriedade' => $this->getTipoPropriedadeAttribute(),
			'tipoVeiculo' => $this->getTipoVeiculoAttribute(),
			'tipoRodado' => $this->getTipoRodadoAttribute(),
			'tipoCarroceria' => $this->getTipoCarroceriaAttribute(),
			'uf' => $this->getUfAttribute(),
			'proprietarioCpf' => $this->getProprietarioCpfAttribute(),
			'proprietarioCnpj' => $this->getProprietarioCnpjAttribute(),
			'proprietarioRntrc' => $this->getProprietarioRntrcAttribute(),
			'proprietarioNome' => $this->getProprietarioNomeAttribute(),
			'proprietarioIe' => $this->getProprietarioIeAttribute(),
			'proprietarioUf' => $this->getProprietarioUfAttribute(),
			'proprietarioTipo' => $this->getProprietarioTipoAttribute(),
        ];
    }
}

File: fontes/frontend/fenix_provider/lib/src/model/cte/cte_rodoviario_veiculo.dart
/*
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [CTE_RODOVIARIO_VEICULO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
import 'dart:convert';


class CteRodoviarioVeiculo {
	int? id;
	int? idCteRodoviario;
	String? codigoInterno;
	String? renavam;
	String? placa;
	int? tara;
	int? capacidadeKg;
	int? capacidadeM3;
	String? tipoPropriedade;
	String? tipoVeiculo;
	String? tipoRodado;
	String? tipoCarroceria;
	String? uf;
	String? proprietarioCpf;
	String? proprietarioCnpj;
	String? proprietarioRntrc;
	String? proprietarioNome;
	String? proprietarioIe;
	String? proprietarioUf;
	String? proprietarioTipo;

	CteRodoviarioVeiculo({
		id,
		idCteRodoviario,
		codigoInterno,
		renavam,
		placa,
		tara,
		capacidadeKg,
		capacidadeM3,
		tipoPropriedade,
		tipoVeiculo,
		tipoRodado,
		tipoCarroceria,
		uf,
		proprietarioCpf,
		proprietarioCnpj,
		proprietarioRntrc,
		proprietarioNome,
		proprietarioIe,
		proprietarioUf,
		proprietarioTipo,
	});

	static List<String> campos = <String>[
		'ID', 
		'CODIGO_INTERNO', 
		'RENAVAM', 
		'PLACA', 
		'TARA', 
		'CAPACIDADE_KG', 
		'CAPACIDADE_M3', 
		'TIPO_PROPRIEDADE', 
		'TIPO_VEICULO', 
		'TIPO_RODADO', 
		'TIPO_CARROCERIA', 
		'UF', 
		'PROPRIETARIO_CPF', 
		'PROPRIETARIO_CNPJ', 
		'PROPRIETARIO_RNTRC', 
		'PROPRIETARIO_NOME', 
		'PROPRIETARIO_IE', 
		'PROPRIETARIO_UF', 
		'PROPRIETARIO_TIPO', 
	];
	
	static List<String> colunas = <String>[
		'Id', 
		'Codigo Interno', 
		'Renavam', 
		'Placa', 
		'Tara', 
		'Capacidade Kg', 
		'Capacidade M 3', 
		'Tipo Propriedade', 
		'Tipo Veiculo', 
		'Tipo Rodado', 
		'Tipo Carroceria', 
		'Uf', 
		'Proprietario Cpf', 
		'Proprietario Cnpj', 
		'Proprietario Rntrc', 
		'Proprietario Nome', 
		'Proprietario Ie', 
		'Proprietario Uf', 
		'Proprietario Tipo', 
	];

	CteRodoviarioVeiculo.fromJson(Map<String, dynamic> jsonDados) {
		id = jsonDados['id'];
		idCteRodoviario = jsonDados['idCteRodoviario'];
		codigoInterno = jsonDados['codigoInterno'];
		renavam = jsonDados['renavam'];
		placa = jsonDados['placa'];
		tara = jsonDados['tara'];
		capacidadeKg = jsonDados['capacidadeKg'];
		capacidadeM3 = jsonDados['capacidadeM3'];
		tipoPropriedade = jsonDados['tipoPropriedade'];
		tipoVeiculo = jsonDados['tipoVeiculo'];
		tipoRodado = jsonDados['tipoRodado'];
		tipoCarroceria = jsonDados['tipoCarroceria'];
		uf = jsonDados['uf'];
		proprietarioCpf = jsonDados['proprietarioCpf'];
		proprietarioCnpj = jsonDados['proprietarioCnpj'];
		proprietarioRntrc = jsonDados['proprietarioRntrc'];
		proprietarioNome = jsonDados['proprietarioNome'];
		proprietarioIe = jsonDados['proprietarioIe'];
		proprietarioUf = jsonDados['proprietarioUf'];
		proprietarioTipo = jsonDados['proprietarioTipo'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonDados = <String, dynamic>{};

		jsonDados['id'] = id ?? 0;
		jsonDados['idCteRodoviario'] = idCteRodoviario ?? 0;
		jsonDados['codigoInterno'] = codigoInterno;
		jsonDados['renavam'] = renavam;
		jsonDados['placa'] = placa;
		jsonDados['tara'] = tara ?? 0;
		jsonDados['capacidadeKg'] = capacidadeKg ?? 0;
		jsonDados['capacidadeM3'] = capacidadeM3 ?? 0;
		jsonDados['tipoPropriedade'] = tipoPropriedade;
		jsonDados['tipoVeiculo'] = tipoVeiculo;
		jsonDados['tipoRodado'] = tipoRodado;
		jsonDados['tipoCarroceria'] = tipoCarroceria;
		jsonDados['uf'] = uf;
		jsonDados['proprietarioCpf'] = proprietarioCpf;
		jsonDados['proprietarioCnpj'] = proprietarioCnpj;
		jsonDados['proprietarioRntrc'] = proprietarioRntrc;
		jsonDados['proprietarioNome'] = proprietarioNome;
		jsonDados['proprietarioIe'] = proprietarioIe;
		jsonDados['proprietarioUf'] = proprietarioUf;
		jsonDados['proprietarioTipo'] = proprietarioTipo;
	
		return jsonDados;
	}
	

	String objetoEncodeJson(CteRodoviarioVeiculo objeto) {
	  final jsonDados = objeto.toJson;
	  return json.encode(jsonDados);
	}
	
}